crates/connector_proxy/src/main.rs

```diff
@@ -114,54 +128,99 @@ async fn signal_handler() {
 }
 
 async fn async_main(
-    image_inspect_json_path: String,
+    image_inspect_json_path: Option<String>,
     proxy_command: ProxyCommand,
 ) -> Result<(), Error> {
-    let image_config = ImageConfig::parse_from_json_file(image_inspect_json_path)?;
-
     // TODO(jixiang): add a check to make sure the proxy_command passed in from commandline is consistent with the protocol inferred from image.
     match proxy_command {
-        ProxyCommand::ProxyFlowCapture(c) => proxy_flow_capture(c, image_config).await,
-        ProxyCommand::ProxyFlowMaterialize(m) => proxy_flow_materialize(m, image_config).await,
+        ProxyCommand::ProxyFlowCapture(c) => proxy_flow_capture(c, image_inspect_json_path).await,
+        ProxyCommand::ProxyFlowMaterialize(m) => {
+            proxy_flow_materialize(m, image_inspect_json_path).await
+        }
+        ProxyCommand::DelayedExecute(ba) => delayed_execute(ba.config_file_path).await,
```

Comment:
Not using a bash script to start the `proxy-process`, b/c we cannot assume the shell script are working for all docker images, (e.g. `distroless`). So I structured it as this additional sub-command of the `connector-proxy`, which basically a separate process that will start the real connector after received some signals from the `connector-proxy` that starts it.